src/frontend/src/binder/expr/function.rs

```diff
@@ -277,9 +285,21 @@ impl Binder {
                     clean_flag = false;
                 }
 
+                // Check for potential recursive calling
+                if self.udf_context.global_count() >= SQL_UDF_MAX_CALLING_DEPTH {
+                    return Err(ErrorCode::BindError(format!(
+                        "function {} calling stack depth limit exceeded",
+                        &function_name
+                    ))
+                    .into());
+                } else {
+                    // Update the status for the global counter
+                    self.udf_context.incr_global_count();
```

Comment:
It seems not right that this counter never decreases?

Reply:
The counter will be reset to 0 after a success binding to a specific sql udf. (i.e., when cleaning the `udf_context`)
```rust
// Clean the `udf_context` & `udf_recursive_context` after inlining,
// which makes sure the subsequent binding will not be affected
if clean_flag {
    self.udf_context.clear();
}
```